src/com/facebook/buck/ide/intellij/IjModuleGraphFactory.java

```diff
@@ -315,6 +325,37 @@ public static IjModuleGraph from(
     return new IjModuleGraph(depsBuilder.build());
   }
 
+  private static IjModule createExtraModuleForCompilerOutput(
+      IjModule module, IjProjectConfig projectConfig, ProjectFilesystem projectFilesystem) {
+    Path projectRootPath = Paths.get(projectConfig.getProjectRoot());
+    Path extraModuleBasePath =
+        projectRootPath
+            .resolve("buck-out")
+            .resolve("extra-intellij-output")
```

Comment:
the logic is already gated on https://github.com/facebook/buck/pull/1429/files/3c9b7a377ec8cbaf2a2d847ca19765603d2440a5#diff-4b53a690d0d3ac329da62c36fcacf204R306 by the buck flag `enable_extra_compiler_output_modules`. is that still necessary?